tests/appsec/test_event_tracking_v2.py

```diff
@@ -0,0 +1,418 @@
+            tag = prefix + "." + key
+            assert tag not in span["meta"], f"Tag {tag} found in span's meta"
+
+    return True
+
+
+@features.event_tracking_sdk_v2
+@scenarios.appsec_ato_sdk
+class Test_UserLoginSuccessEventV2_Tags:
+    """Test tags created in AppSec User Login Success Event SDK v2"""
+
+    def get_user_login_success_tags_validator(self, login, user_id, metadata=None, unexpected_metadata=None):
+        def validate(span):
+            expected_tags = {
+                "appsec.events.users.login.success.usr.login": login,
+                "appsec.events.users.login.success.usr.id": user_id,
+                "usr.id": user_id,
+                "appsec.events.users.login.success.track": "true",
+                "_dd.appsec.events.users.login.success.sdk": "true",
+                "http.client_ip": "1.2.3.4",
+                "_dd.appsec.user.collection_mode": "sdk",
+            }
+
+            return validate_tags_and_metadata(
+                span, "appsec.events.users.login.success", expected_tags, metadata, unexpected_metadata
+            )
+
+        return validate
+
+    def setup_user_login_success_event_strings_metadata(self):
+        headers = {
+            "X-Forwarded-For": "1.2.3.4",
+        }
+
+        metadata = {"metadata0": "value0", "metadata1": "value1"}
+
+        data = {"login": LOGIN_SAFE, "user_id": USER_ID_SAFE, "metadata": metadata}
+
+        self.r = weblog.post("/user_login_success_event_v2", json=data, headers=headers)
+
+    def test_user_login_success_event_strings_metadata(self):
+        # Call the user login success SDK and validate tags
+
+        metadata = {"metadata0": "value0", "metadata1": "value1"}
+
+        interfaces.library.validate_spans(
+            self.r, self.get_user_login_success_tags_validator(LOGIN_SAFE, USER_ID_SAFE, metadata=metadata)
+        )
+
+    def setup_user_login_success_event_multi_type_metadata(self):
+        headers = {
+            "X-Forwarded-For": "1.2.3.4",
+        }
+
+        metadata = {"metadata0": "value0", "metadata_number": 123, "metadata_boolean": True}
+
+        data = {"login": LOGIN_SAFE, "user_id": USER_ID_SAFE, "metadata": metadata}
+
+        self.r = weblog.post("/user_login_success_event_v2", json=data, headers=headers)
+
+    def test_user_login_success_event_multi_type_metadata(self):
+        # Call the user login success SDK and validate tags
+
+        metadata = {"metadata0": "value0", "metadata_number": "123", "metadata_boolean": "true"}
+
+        interfaces.library.validate_spans(
+            self.r, self.get_user_login_success_tags_validator(LOGIN_SAFE, USER_ID_SAFE, metadata=metadata)
+        )
+
+    def setup_user_login_success_event_no_metadata(self):
+        headers = {
+            "X-Forwarded-For": "1.2.3.4",
+        }
+
+        data = {"login": LOGIN_SAFE, "user_id": USER_ID_SAFE}
+
+        self.r = weblog.post("/user_login_success_event_v2", data=data, headers=headers)
```

Comment:
Can we always use json for the payload?, it would be painful to maintain both versions in all Java weblogs :cry: